src/qt/TailsOS.cpp

```diff
@@ -0,0 +1,75 @@
+#include <QRegExp>
+#include <QMessageBox>
+#include <QPixmap>
+#include <QTranslator>
+
+#include "TailsOS.h"
+#include "utils.h"
+
+bool tailsUsePersistence = false;
+
+bool TailsOS::detect()
+{
+    if (!fileExists("/etc/os-release"))
+        return false;
+
+    QByteArray data = fileOpen("/etc/os-release");
+    QRegularExpression re("TAILS_PRODUCT_NAME=\"Tails\"");
+    QRegularExpressionMatch os_match = re.match(data);
+    bool matched = os_match.hasMatch();
+#ifdef QT_DEBUG
+    if (matched)
+        qDebug() << "Tails OS detected";
+#endif
+    return matched;
+}
+
+bool TailsOS::detectDataPersistence()
+{
+    return QDir(QDir::homePath() + "/Persistent").exists();
+}
+
+bool TailsOS::detectDotPersistence()
+{
+    return QDir("/live/persistence/TailsData_unlocked/dotfiles").exists();
+}
+
+void TailsOS::showDataPersistenceDisabledWarning()
+{
+    QMessageBox msgBox;
```

Comment:
The design of these doesn’t match with the rest of the GUI. While we use `MessageDialog` in some minor places, it’s just because they haven’t been replaced yet, the user doesn’t see them during normal use.

What’s your opinion on this?

Reply:
I think it is fine, this is only for Tails - before QML has started.

Reply:
Right, it’s before QML even started. Ok then.